docs/source/introducing_v4.rst

```diff
@@ -67,6 +67,38 @@ In pyQuil v4, Gateway is enabled by default and it is generally recommended to k
     result = qc.qam.execution_options = execution_options
 
 
+Accessing Raw Execution Data
+----------------------------
+
+In previous versions of pyQuil, readout data was always returned as a mapping of memory regions to rectangular matrices
+that contained one value per memory reference, per shot. However, it shouldn't be assumed that readout data will always
+fit this shape. For example, programs that use mid-circuit measurement or dynamic control flow can emit a different
```

Comment:
```suggestion
fit this shape. For example, programs that reuse qubits or use dynamic control flow can emit a different
```

MCM itself (i.e., measuring before program end) should not change the results shape unless the qubit is reused